src/Commands/core/BrowseCommands.php

```diff
@@ -28,11 +31,20 @@ class BrowseCommands extends DrushCommands
      */
     public function browse($path = '', $options = ['browser' => null])
     {
+        // TODO: Remove 2nd branch when no longer needed.
+        if ($this->hasSiteAliasManager()) {
+            $aliasRecord = $this->siteAliasManager()->getSelf();
+            $is_remote = $aliasRecord->isRemote();
+            $site_record = $aliasRecord->exportConfig()->get('options', []);
+        }
+        else {
+            $alias = drush_get_context('DRUSH_TARGET_SITE_ALIAS');
+            $is_remote = drush_sitealias_is_remote_site($alias);
+            $site_record = drush_sitealias_get_record($alias);
+        }
         // Redispatch if called against a remote-host so a browser is started on the
         // the *local* machine.
-        $alias = drush_get_context('DRUSH_TARGET_SITE_ALIAS');
-        if (drush_sitealias_is_remote_site($alias)) {
-            $site_record = drush_sitealias_get_record($alias);
+        if ($is_remote) {
             $return = drush_invoke_process($site_record, 'browse', [$path], drush_redispatch_get_options(), array('integrate' => true));
```

Comment:
Ah, I see, we are still using `drush_redispatch_get_options()` here. Ideally this method goes away once all of the commands are ported. Maybe use `$options` here?  As previously mentioned, `$options` may have items that do not appear in `drush_redispatch_get_options()` (default values for options that do not appear on the commandline).

Maybe we need to add a method to the PreflightArgs object that contains the original user options from the commandline. Then the question is, do we inject this only for commands that need it, or add an accessor to DrushCommands?

Reply:
I changed this to $options but now the receiving Drush (i.e. the remote in the $site_record) is throwing an error (it happens to be  Drush9 beta3 in this case) - `Unknown options: --version, --ansi, --no-ansi, --no-interaction.  See drush help browse for available options. To suppress this error, add the option --strict=0. `. So, backend seems to be passing along too many options. Thoughts?

Reply:
Yes, that is exactly what I was referring to in my comment above. Symfony adds in any option with a default value to `$options` (well, Symfony puts it in `$input`, and annotated-commands translates that into `$options`). We need a way for commands that need to redispatch to inject the original commandline arguments. Or perhaps we create a redispatch object that has the original commandline arguments.